genestack_client/genestack_connection.py

```diff
@@ -151,20 +150,26 @@ def logout(self):
         """
         self.application('genestack/signin').invoke('signOut')
 
-    def open(self, path, data=None, follow=True):
+    def open(self, path, data=None, follow=True, headers=None):
         """
         Sends data to a URL. The URL is the concatenation of the server URL and "path".
 
         :param path: part of URL that is added to self.server_url
         :param data: dict of parameters, file-like objects or strings
         :param follow: should we follow a redirection if any?
+        :param headers: additional headers as list of pairs
+        :type headers: list[tuple[str]]
         :return: response
         :rtype: urllib.addinfourl
         """
         if data is None:
             data = ''
         elif isinstance(data, dict):
             data = urllib.urlencode(data)
+        headers = list(headers)
+        headers.append(('gs-extendSession', 'true'))
```

Comment:
This header was not touched.  

We have 3 headers in total, but they never present all three in same file :)

`gs-extendSession`, used in `python` and `js`,  handled outside `InvokeController`.
`Genestack-ExternalCall`  used in `js` and `InvokeController`
Genestack-Trace used in `python` and `InvokeController`